src/autoload/Global.gd

```diff
@@ -1,11 +1,63 @@
 extends Node
 
 
+const CONFIG_PATH := "user://config.cfg"
+
+var config := {
+	"name": "Guest",
+	"favorite_color": Color(255, 0, 0, 255),
+	"address": "localhost",
+	"port": 8380,
+}
+
+
 func _ready() -> void:
 	randomize()
+	load_config()
+	pause_mode = Node.PAUSE_MODE_PROCESS
+
+
+# When the user quits the game, save the game before the engine fully quits
+func _notification(what: int) -> void:
+	if what == MainLoop.NOTIFICATION_WM_QUIT_REQUEST:
+		save_config()
+		get_tree().quit()
 
 
 func _input(event: InputEvent) -> void:
 	if event.is_action_pressed("toggle_fullscreen"):
 		OS.window_fullscreen = not OS.window_fullscreen
 		get_tree().set_input_as_handled()
+
+
+func load_config():
+	var config_file := File.new()
+	var error := config_file.open(CONFIG_PATH, File.READ)
+	if error:
+		push_warning("Could not open config file for reading")
+		return
```

Comment:
Do we want to do a check (something like os.path.exists(path)) and just return early as well? That way if we actually can't read the file we get a debugger error, but otherwise it doesn't complain?

Reply:
Not a big deal but might as well